tiledb/array.py

```diff
@@ -19,20 +18,21 @@ class DenseArray(DenseArrayImpl):
     and assignment.
     """
 
-    _mixin_init = False
-
     def __new__(cls, *args, **kwargs):
-        if not cls._mixin_init:
-            # must set before importing, because import is not thread-safe
-            #   https://github.com/TileDB-Inc/TileDB-Py/issues/244
-            cls._mixin_init = True
-            try:
-                from tiledb.cloud import cloudarray
-
-                DenseArray.__bases__ = DenseArray.__bases__ + (cloudarray.CloudArray,)
-            except ImportError:
-                pass
-
+        try:
+            from tiledb.cloud import cloudarray
+
+            cls.__bases__ += (cloudarray.CloudArray,)
```

Comment:
Looking at this again, I was slightly wrong at first: we do want to do `DenseArray.__bases__` here instead of `cls.__bases__`. (For more details see the log in the other comment.)

Reply:
You are technically correct, [the best kind of correct](https://www.youtube.com/watch?v=0ZEuWJ4muYc); in practice though I believe `DenseArray/SparseArray` are effectively "final", i.e. they're not meant to be subclassed so `cls is DenseArray`.